src/test/java/org/folio/search/controller/IndexManagementIT.java

```diff
@@ -191,4 +239,30 @@ private static String reindexUnexpectedResource(String resource) {
     return REINDEX_UNEXPECTED_RESOURCE.formatted(resource);
   }
 
+  private static Stream<Arguments> requestBuilderDataProvider() {
+    return Stream.of(
+      arguments(
+        post(ApiEndpoints.reindexPath())
+          .content(asJsonString(new ReindexRequest().resourceName(LINKED_DATA_WORK)))
+          .headers(defaultHeaders())
+          .header(XOkapiHeaders.URL, okapi.getOkapiUrl())
+          .contentType(MediaType.APPLICATION_JSON)
+      ),
+      arguments(
+        post(ApiEndpoints.reindexPath())
+          .content(asJsonString(new ReindexRequest().resourceName(LINKED_DATA_WORK).recreateIndex(true)))
+          .headers(defaultHeaders(MEMBER_TENANT_ID))
+          .header(XOkapiHeaders.URL, okapi.getOkapiUrl())
+          .contentType(MediaType.APPLICATION_JSON)
+      ),
+      arguments(
+        post(ApiEndpoints.reindexPath())
+          .content(asJsonString(new ReindexRequest().resourceName(AUTHORITY).recreateIndex(true)))
+          .headers(defaultHeaders())
+          .header(XOkapiHeaders.URL, okapi.getOkapiUrl())
+          .contentType(MediaType.APPLICATION_JSON)
```

Comment:
can extract requestBuilder creation to separate method to avoid code duplication

Reply:
or just having a request builder in the test method and in the argument provider having only ReindexRequest objects.

Reply:
Providing only ```ReindexRequest``` objects will not work here because I use ```MEMBER_TENANT_ID``` in one of the cases.

Reply:
Extracted ```requestBuilder``` creation to a separate method.